src/components/vanilla/charts/CompareLineChart/index.tsx

```diff
@@ -0,0 +1,288 @@
+          }
+        },
+        period: {
+          min: bounds.period?.from?.toJSON(),
+          max: bounds.period?.to?.toJSON(),
+          grid: {
+            display: false
+          },
+          type: 'time',
+          time: {
+            round: props.granularity,
+            isoWeekday: true,
+            displayFormats: {
+              month: 'MMM',
+              day: 'd MMM',
+              week: 'd MMM',
+              hour: 'eee HH:mm',
+              minute: 'eee HH:mm',
+              second: 'HH:mm:ss'
+            },
+            unit: props.granularity
+          },
+          title: {
+            display: !!props.xAxisTitle,
+            text: props.xAxisTitle
+          }
+        },
+        comparison: {
+          min: bounds.comparison?.from?.toJSON(),
+          max: bounds.comparison?.to?.toJSON(),
+          display: false,
+          grid: {
+            display: false
+          },
+          type: 'time',
+          time: {
+            round: props.granularity,
+            isoWeekday: true,
+            displayFormats: {
+              month: 'MMM',
+              day: 'd MMM',
+              week: 'd MMM',
+              hour: 'eee HH:mm',
+              minute: 'eee HH:mm',
+              second: 'HH:mm:ss'
+            },
+            unit: props.granularity
+          },
+          title: {
+            display: false
+          }
+        }
+      },
+      animation: {
+        duration: 400,
+        easing: 'linear'
+      },
+      plugins: {
+        legend: {
+          display: props.showLegend,
+          position: 'bottom',
+          labels: {
+            usePointStyle: true,
+            boxHeight: 8
+          }
+        },
+        datalabels: {
+          align: 'top',
+          display: props.showLabels ? 'auto' : false,
+          formatter: (v) => {
+            const val = v ? formatValue(v.y, { type: 'number', dps: props.dps }) : null;
+            return val;
+          }
+        }
+      }
+    };
+  }, [props]);
+
+  useEffect(() => {
+    ChartJS.defaults.plugins.tooltip.callbacks.title = (lines: any[]) => {
+      const definedFormat = {
+        quarter: 'MMM yyyy',
+        month: 'MMM',
+        day: 'd MMM',
+        week: 'd MMM',
+        hour: 'eee HH:mm',
+        minute: 'eee HH:mm',
+        second: 'HH:mm:ss'
+      };
+
+      const linesList = lines.map((line) => line?.raw?.x?.valueOf());
+      const set = [...new Set(linesList)];
+      return set.map((date) => format(new Date(date), definedFormat[props.granularity]))
+        .filter((v) => !!v);
+    };
+  }, [props.granularity]);
+
+  return (
+    <Container className="overflow-y-hidden" title={props.title} results={props.results}>
+      <Line height="100%" options={chartOptions} data={chartData} />
+    </Container>
+  );
+};
+
+// Convert hex to rgb and add opacity. Used when a color-fill is applied beneath the chart line(s).
+function hexToRgb(hex: string, alpha?: number): string {
```

Comment:
we already have this function on the utils, lets reuse it

src/components/util/hexToRgb.ts

Reply:
@mad-raccoon done.